src/EnergyPlus/PurchasedAirManager.cc

```diff
@@ -233,876 +233,953 @@ void GetPurchasedAir(EnergyPlusData &state)
+            PurchAir.MinCoolSuppAirHumRat = s_ip->getRealFieldValue(fields, schemaProps, "minimum_cooling_supply_air_humidity_ratio");
+            cAlphaFieldName = "Heating Limit";
+            fieldValue = s_ip->getAlphaFieldValue(fields, schemaProps, "heating_limit");
+            PurchAir.HeatingLimit = static_cast<LimitType>(getEnumValue(limitTypeNamesUC, Util::makeUPPER(fieldValue)));
+            switch (PurchAir.HeatingLimit) {
 
             case LimitType::None: {
             } break;
 
             case LimitType::FlowRate: {
-                if (s_ipsc->lNumericFieldBlanks(5)) {
-                    PurchAir(PurchAirNum).HeatingLimit = LimitType::None;
+                if (fields.find("maximum_heating_air_flow_rate") == fields.end()) {
+                    PurchAir.HeatingLimit = LimitType::None;
+                } else {
+                    PurchAir.HeatingLimit = LimitType::FlowRate;
                 }
             } break;
 
             case LimitType::Capacity: {
-                if (s_ipsc->lNumericFieldBlanks(6)) {
-                    PurchAir(PurchAirNum).HeatingLimit = LimitType::None;
+                if (fields.find("maximum_sensible_heating_capacity") == fields.end()) {
+                    PurchAir.HeatingLimit = LimitType::None;
+                } else {
+                    PurchAir.HeatingLimit = LimitType::Capacity;
                 }
             } break;
 
             case LimitType::FlowRateAndCapacity: {
-                if (s_ipsc->lNumericFieldBlanks(5) && s_ipsc->lNumericFieldBlanks(6)) {
-                    PurchAir(PurchAirNum).HeatingLimit = LimitType::None;
-                } else if (s_ipsc->lNumericFieldBlanks(5)) {
-                    PurchAir(PurchAirNum).HeatingLimit = LimitType::Capacity;
-                } else if (s_ipsc->lNumericFieldBlanks(6)) {
-                    PurchAir(PurchAirNum).HeatingLimit = LimitType::FlowRate;
+                if ((fields.find("maximum_heating_air_flow_rate") == fields.end()) &&
+                    (fields.find("maximum_sensible_heating_capacity") == fields.end())) {
+                    PurchAir.HeatingLimit = LimitType::None;
+                } else if (fields.find("maximum_heating_air_flow_rate") == fields.end()) {
+                    PurchAir.HeatingLimit = LimitType::Capacity;
+                } else if (fields.find("maximum_sensible_heating_capacity") == fields.end()) {
+                    PurchAir.HeatingLimit = LimitType::FlowRate;
+                } else {
+                    PurchAir.HeatingLimit = LimitType::FlowRateAndCapacity;
                 }
             } break;
 
             default: {
-                ShowSevereInvalidKey(state, eoh, s_ipsc->cAlphaFieldNames(6), s_ipsc->cAlphaArgs(6));
+                ShowSevereInvalidKey(state, eoh, cAlphaFieldName, fieldValue, "Valid entries are None, FlowRate, Capacity, or FlowRateAndCapacity.");
                 ErrorsFound = true;
             } break;
             }
```

Comment:
This code, lines 369 - 407, is not needed since this field has a default (i.e., this field will not be blank) and the enum was set at line 368. Same comment for Cooling Limit below.

    A6 , \field Heating Limit
       \type choice
       \key NoLimit
       \key LimitFlowRate
       \key LimitCapacity
       \key LimitFlowRateAndCapacity
       \default NoLimit

Reply:
I take that back. The fields being checked here have `\default autosizable` so none of the checks will `== fields.end()`. This code is redundant.